src/cockpit/389-console/src/lib/plugins/attributeUniqueness.jsx

```diff
@@ -439,49 +527,75 @@ class AttributeUniqueness extends React.Component {
 
-        // Delete attributes if the user set an empty value to the field
-        if (!(action === "add" && attrNames.length === 0)) {
+        if (action === "add") {
             cmd = [...cmd, "--attr-name"];
             if (attrNames.length !== 0) {
                 for (const value of attrNames) {
                     cmd = [...cmd, value];
                 }
-            } else if (action === "add") {
-                cmd = [...cmd, ""];
-            } else {
-                cmd = [...cmd, "delete"];
             }
-        }
-
-        if (!(action === "add" && subtrees.length === 0)) {
-            cmd = [...cmd, "--subtree"];
             if (subtrees.length !== 0) {
+                cmd = [...cmd, "--subtree"];
                 for (const value of subtrees) {
                     cmd = [...cmd, value];
                 }
-            } else if (action === "add") {
-                cmd = [...cmd, ""];
-            } else {
-                cmd = [...cmd, "delete"];
             }
-        }
-
-        cmd = [...cmd, "--top-entry-oc"];
-        if (topEntryOc.length !== 0) {
-            cmd = [...cmd, topEntryOc];
-        } else if (action === "add") {
-            cmd = [...cmd, ""];
-        } else {
-            cmd = [...cmd, "delete"];
-        }
-
-        cmd = [...cmd, "--subtree-entries-oc"];
-        if (subtreeEnriesOc.length !== 0) {
-            cmd = [...cmd, subtreeEnriesOc];
-        } else if (action === "add") {
-            cmd = [...cmd, ""];
+            if (excludeSubtrees.length !== 0) {
+                cmd = [...cmd, "--exclude-subtree"];
+                for (const value of excludeSubtrees) {
+                    cmd = [...cmd, value];
+                }
+            }
+            if (topEntryOc.length !== 0) {
+                cmd = [...cmd, "--top-entry-oc", topEntryOc];
+            }
+            if (subtreeEnriesOc.length !== 0) {
+                cmd = [...cmd, "--subtree-entries-oc", subtreeEnriesOc];
+            }
         } else {
-            cmd = [...cmd, "delete"];
+            // Set/edit
+            if (!listsEqual(this.state['attrNames'], this.state['_attrNames'])) {
+                cmd = [...cmd, "--attr-name"];
+                for (const value of attrNames) {
+                    cmd = [...cmd, value];
+                }
+            }
+            if (!listsEqual(this.state['subtrees'], this.state['_subtrees'])) {
+                cmd = [...cmd, "--subtree"];
+                if (subtrees.length === 0) {
+                    // Remove all values
+                    cmd = [...cmd, "delete"];
+                } else {
+                    for (const value of subtrees) {
+                        cmd = [...cmd, value];
+                    }
+                }
+            }
+            if (!listsEqual(this.state['excludeSubtrees'], this.state['_excludeSubtrees'])) {
+                cmd = [...cmd, "--exclude-subtree"];
+                if (exclude_subtrees.length === 0) {
```

Comment:
Should it be `excludeSubtrees` instead of `exclude_subtrees`?

Reply:
> Should it be `excludeSubtrees` instead of `exclude_subtrees`?

Nice catch!